04-onion-routing.md

```diff
@@ -1407,6 +1438,117 @@ The _origin node_:
   - MAY use the data specified in the various failure types for debugging
   purposes.
 
+## Recommendations for Reputation
+
+Local reputation is used by forwarding nodes to decide whether to endorse a HTLC
+on their outgoing link. Nodes MAY use any metric of their choosing to classify
+a peer as having "good" reputation, though a poor choice of reputation scoring
+metric may affect their downstream reputation.
+
+Peers build reputation by forwarding successful HTLCs that resolve quickly, and
+lose reputation if they endorse failing or slow-resolving HTLCs. Reputation is
+only _negatively_ affected if an endorsed HTLC resolves undesirably, to hold
+nodes accountable for their endorsement signal while still allowing them to 
+forward unendorsed HTLCs that they are not certain about.
```

Comment:
I'm not sure this statement is true, because of the following scenario (let me know if I'm missing the point entirely):

- A wants to send a payment to D: A -> B -> C -> D
- the A -> B and B -> C channels are empty: A and B both endorse the HTLC
- the C -> D channel is full (because of payments coming from unrelated nodes, e.g. E -> C -> D)
- when C receives the HTLC, even it's fully endorsed and reputable, C has to fail the payment
- when B receives the failure, B will then decrease A's reputation

Is the negative impact on A an issue? Is it something that can be abused? Can we do something about it (should we)?

Reply:
A quick failure will not harm your reputation.
Only slow resolving endorsed HTLC can harm your reputation.

Reply:
Ok, I believe that's one of the main differences between @thomash-acinq's proposal and this one, it's hard to evaluate which is the right choice.